casper/src/main/scala/coop/rchain/casper/SafetyOracle.scala

```diff
@@ -119,6 +119,24 @@ sealed abstract class SafetyOracleInstances {
           }
         } yield mainParentWeightMap
 
+      def findMaximumClique(edges: List[(Validator, Validator)],
+                            candidates: Map[Validator, Int]): F[(List[Validator], Int)] =
+        Clique
+          .findCliquesRecursive(edges)
+          .foldLeft((List[Validator](), 0)) {
+            case ((maxClique, maxWeight), clique) => {
+              val weight = clique.map(candidates.getOrElse(_, 0)).sum
+              if (weight > maxWeight) {
+                (clique, weight)
+              } else if (weight == maxWeight && clique.size > maxClique.size) {
+                (clique, weight)
+              } else {
+                (maxClique, maxWeight)
+              }
+            }
+          }
+          .pure[F]
```

Comment:
There is no need to close this over F. Just have this as a pure function.

Reply:
Done